dtfs-central-api/src/v1/controllers/utilisation-report-service/get-utilisation-report-reconciliation-details-by-id.controller/helpers/map-fee-record-payment-entity-groups-to-fee-record-payment-groups.test.ts

```diff
@@ -53,64 +82,106 @@ describe('get-utilisation-report-reconciliation-details-by-id.controller helpers
         ];
 
         // Act
-        const result = mapFeeRecordPaymentEntityGroupsToFeeRecordPaymentGroups(groups);
+        const result = await mapFeeRecordPaymentEntityGroupsToFeeRecordPaymentGroups(groups);
 
         // Assert
         result.forEach((group, index) => {
           expect(group.status).toBe(groups[index].feeRecords[0].status);
         });
       });
 
-      it('sets the totalReportedPayments to the same value as the fee record reported payments', () => {
+      it('sets the totalReportedPayments to the same value as the fee record reported payments', async () => {
         // Arrange
         const groups = [createFeeRecordEntityPaymentGroupForSingleFeeRecord(1, 'TO_DO')];
 
         // Act
-        const result = mapFeeRecordPaymentEntityGroupsToFeeRecordPaymentGroups(groups);
+        const result = await mapFeeRecordPaymentEntityGroupsToFeeRecordPaymentGroups(groups);
 
         // Assert
         expect(result[0].totalReportedPayments).toEqual({ currency, amount });
       });
 
-      it('sets the paymentsReceived to null', () => {
+      it('sets the paymentsReceived to null', async () => {
         // Arrange
         const groups = [createFeeRecordEntityPaymentGroupForSingleFeeRecord(1, 'TO_DO')];
 
         // Act
-        const result = mapFeeRecordPaymentEntityGroupsToFeeRecordPaymentGroups(groups);
+        const result = await mapFeeRecordPaymentEntityGroupsToFeeRecordPaymentGroups(groups);
 
         // Assert
         expect(result[0].paymentsReceived).toBeNull();
       });
 
-      it('sets the totalPaymentsReceived to null', () => {
+      it('sets the totalPaymentsReceived to null', async () => {
         // Arrange
         const groups = [createFeeRecordEntityPaymentGroupForSingleFeeRecord(1, 'TO_DO')];
 
         // Act
-        const result = mapFeeRecordPaymentEntityGroupsToFeeRecordPaymentGroups(groups);
+        const result = await mapFeeRecordPaymentEntityGroupsToFeeRecordPaymentGroups(groups);
 
         // Assert
         expect(result[0].totalPaymentsReceived).toBeNull();
       });
+
+      it.each(NON_RECONCILED_FEE_RECORD_STATUSES)('sets the dateReconciled and reconciledByUser to null when the fee record status is %s', async (status) => {
+        // Arrange
+        const group: FeeRecordPaymentEntityGroup = {
+          feeRecords: [
+            FeeRecordEntityMockBuilder.forReport(aUtilisationReport())
+              .withStatus(status)
+              .withDateReconciled(new Date('2024')) // this value would normally not be defined, but is being defined here for testing purposes
+              .withReconciledByUserId('abc123') // this value would normally not be defined, but is being defined here for testing purposes
```

Comment:
For my understanding, why would these values be undefined and what's the value in having them defined in the test?

Do we have tests for when either or both of these are undefined?

Reply:
Those values would normally only ever be defined if the fee record status is `RECONCILED` - any other time it would be `null`

Reply:
I am confused,

- The function will return these properties if they are returned by `getReconciliationDataForFeeRecords`.
- You mention they would only be defined if the status is `RECONCILED`, I can't see in the test where this is indicated that it's `RECONCILED`?

Do we have tests for when either or both of these are undefined?